netket/graph/common_lattices.py

```diff
@@ -0,0 +1,403 @@
+             PBC/OBC depending on the corresponding entry of `pbc`.
+
+    Example:
+        Construct a BCC lattice with 3×3×3 primitive unit cells:
+
+        >>> from netket.graph import BCC
+        >>> g = BCC(extent=[3,3,3])
+        >>> print(g.n_nodes)
+        27
+    """
+    basis = [[-0.5, 0.5, 0.5], [0.5, -0.5, 0.5], [0.5, 0.5, -0.5]]
+    # determine if full point group is realised by the simulation box
+    point_group = cubic.Oh() if np.all(pbc) else None
+    return Lattice(basis_vectors=basis, extent=extent, pbc=pbc, point_group=point_group)
+
+
+def FCC(extent: Sequence[int], *, pbc: Union[bool, Sequence[bool]] = True) -> Lattice:
+    """Constructs an FCC lattice of a given spatial extent.
+    Periodic boundary conditions can also be imposed
+    Sites are returned at the Bravais lattice points.
+
+    Arguments:
+        extent: Number of primitive unit cells along each direction, needs to be an array of length 3
+        pbc: If `True`, the lattice will have periodic boundary conditions (PBC);
+             if `False`, the lattice will have open boundary conditions (OBC).
+             This parameter can also be a list of booleans with same length as
+             the parameter `length`, in which case each dimension will have
+             PBC/OBC depending on the corresponding entry of `pbc`.
+
+    Example:
+        Construct an FCC lattice with 3×3×3 primitive unit cells:
+
+        >>> from netket.graph import FCC
+        >>> g = FCC(extent=[3,3,3])
+        >>> print(g.n_nodes)
+        27
+    """
+    basis = [[0, 0.5, 0.5], [0.5, 0, 0.5], [0.5, 0.5, 0]]
+    # determine if full point group is realised by the simulation box
+    point_group = cubic.Oh() if np.all(pbc) else None
+    return Lattice(basis_vectors=basis, extent=extent, pbc=pbc, point_group=point_group)
+
+
+def Diamond(
+    extent: Sequence[int], *, pbc: Union[bool, Sequence[bool]] = True
+) -> Lattice:
+    """Constructs a diamond lattice of a given spatial extent.
+    Periodic boundary conditions can also be imposed.
+
+    Sites are returned at the 8a Wyckoff positions of the FCC lattice
+    ([000], [1/4,1/4,1/4], and translations thereof).
+
+    Arguments:
+        extent: Number of primitive unit cells along each direction, needs to be an array of length 3
+        pbc: If `True`, the lattice will have periodic boundary conditions (PBC);
+             if `False`, the lattice will have open boundary conditions (OBC).
+             This parameter can also be a list of booleans with same length as
+             the parameter `length`, in which case each dimension will have
+             PBC/OBC depending on the corresponding entry of `pbc`.
+
+    Example:
+        Construct a diamond lattice with 3×3×3 primitive unit cells:
+
+        >>> from netket.graph import Diamond
+        >>> g = Diamond(extent=[3,3,3])
+        >>> print(g.n_nodes)
+        54
+    """
+    basis = [[0, 0.5, 0.5], [0.5, 0, 0.5], [0.5, 0.5, 0]]
```

Comment:
Are the volumes of these unit cells chosen according to a particular convention? Otherwise they should probably have unit volume

Reply:
Yes, FCC and BCC lattices in particular are defined to have a unit-length cubic unit cell. (Please don't make the hexagonal unit cells unit volume.)